src/JWT/Serializers/JsonNetSerializer.cs

```diff
@@ -8,22 +9,34 @@ namespace JWT.Serializers
     /// </summary>
     public sealed class JsonNetSerializer : IJsonSerializer
     {
+        private readonly JsonSerializer _serializer;
+
         /// <summary>
-        /// Serialize the given object.
+        /// Creates a new instance of <see cref="JsonNetSerializer" />.
         /// </summary>
-        /// <param name="obj">The object to serialize.</param>
-        /// <returns></returns>
-        public string Serialize(object obj)
+        /// <remarks>Uses <see cref="JsonSerializer.CreateDefault()" /> as internal serializer.</remarks>
+        public JsonNetSerializer()
+            : this(JsonSerializer.CreateDefault())
         {
-            return JObject.FromObject(obj).ToString(Formatting.None);
+
         }
 
         /// <summary>
-        /// Deserialize the given string.
+        /// Creates a new instance of <see cref="JsonNetSerializer" />.
         /// </summary>
-        /// <typeparam name="T">The type to deserialize the string to.</typeparam>
-        /// <param name="json">The JSON to be deserialized.</param>
-        /// <returns></returns>
+        /// <param name="serializer">Internal <see cref="JsonSerializer" /> to use for serialization.</param>
+        public JsonNetSerializer(JsonSerializer serializer)
+        {
+            _serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
+        }
+
+        /// <inheritdoc />
+        public string Serialize(object obj)
+        {
+            return JObject.FromObject(obj, _serializer).ToString(Formatting.None);

```

Comment:
Oh . that said ...

I couldn't find an online token decoder that RESPECTED the json formatting which was encoded into the JWT. All the online decoders I _think_ formatted the json nicely for me (which makes all of this testing, hardish)